From 3f74a606eded9289b70383f6106c1a0eaeca4bf8 Mon Sep 17 00:00:00 2001
From: Haibo Chen <haibo.chen@nxp.com>
Date: Fri, 8 Jan 2016 20:15:05 +0800
Subject: [PATCH 1/2] MLK-12140 ARM: dts: imx7d-sdb-epdc: add EPDC touch
 support for imx7d-sdb

commit  8bd378a7931c85dbda3dc4b6efd1f9e8af736a28 from
https://source.codeaurora.org/external/imx/linux-imx.git

EPDC board has touch screen, this patch add touch support on imx7d-sdb TO1.1
board, due to the touch pin conflict with UART5 and GPMI, so disable UART5 in
imx7d-sdb-epdc.dtsi. For GPMI, it default disabled in imx7d-sdb.dts, so do not
impact touch.

Signed-off-by: Haibo Chen <haibo.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx7d-sdb-epdc.dtsi | 19 +++++++++++++++++++
 arch/arm/boot/dts/imx7d-sdb.dts       |  8 ++++++++
 2 files changed, 27 insertions(+)

diff --git a/arch/arm/boot/dts/imx7d-sdb-epdc.dtsi b/arch/arm/boot/dts/imx7d-sdb-epdc.dtsi
index f283e4e9fc91..765fdc4bba72 100644
--- a/arch/arm/boot/dts/imx7d-sdb-epdc.dtsi
+++ b/arch/arm/boot/dts/imx7d-sdb-epdc.dtsi
@@ -37,3 +37,22 @@
 &sim1 {
 	status = "disabled";
 };
+
+&uart5 {
+	status = "disabled";
+};
+
+&i2c3 {
+	elan@10 {
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_epdc_elan_touch>;
+		compatible = "elan,elan-touch";
+		reg = <0x10>;
+		interrupt-parent = <&gpio6>;
+		interrupts = <12 IRQ_TYPE_EDGE_FALLING>;
+		gpio_elan_cs = <&gpio6 13 0>;
+		gpio_elan_rst = <&gpio6 15 0>;
+		gpio_intr = <&gpio6 12 0>;
+		status = "okay";
+	};
+};
diff --git a/arch/arm/boot/dts/imx7d-sdb.dts b/arch/arm/boot/dts/imx7d-sdb.dts
index 9da5e9873583..cb21b37f9284 100644
--- a/arch/arm/boot/dts/imx7d-sdb.dts
+++ b/arch/arm/boot/dts/imx7d-sdb.dts
@@ -784,6 +784,14 @@
 			>;
 		};
 
+		pinctrl_epdc_elan_touch: epdc_elan_touch_grp {
+			fsl,pins = <
+				MX7D_PAD_SAI1_RX_DATA__GPIO6_IO12 0x59
+				MX7D_PAD_SAI1_TX_DATA__GPIO6_IO15 0x1b
+				MX7D_PAD_SAI1_TX_BCLK__GPIO6_IO13 0x80000000
+			>;
+		};
+
 		pinctrl_hog_1: hoggrp-1 {
 			fsl,pins = <
 				MX7D_PAD_EPDC_BDR0__GPIO2_IO28	0x59 /* headphone detect */
-- 
2.17.1

